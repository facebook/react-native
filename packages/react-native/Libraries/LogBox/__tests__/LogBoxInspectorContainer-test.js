/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 * @oncall react_native
 */

'use strict';

const render = require('../../../jest/renderer');
const LogBoxLog = require('../Data/LogBoxLog').default;
const {
  _LogBoxNotificationContainer: LogBoxNotificationContainer,
} = require('../LogBoxNotificationContainer');
const React = require('react');

// Mock `LogBoxLogNotification` because we are interested in snapshotting the
// behavior of `LogBoxNotificationContainer`, not `LogBoxLogNotification`.
jest.mock('../UI/LogBoxNotification', () => ({
  __esModule: true,
  default: 'LogBoxLogNotification',
}));

describe('LogBoxNotificationContainer', () => {
  it('should render null with no logs', async () => {
    const output = await render.create(
      <LogBoxNotificationContainer selectedLogIndex={-1} logs={[]} />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render null with no selected log and disabled', async () => {
    const output = await render.create(
      <LogBoxNotificationContainer
        isDisabled
        selectedLogIndex={-1}
        logs={[
          new LogBoxLog({
            level: 'warn',
            isComponentError: false,
            message: {
              content: 'Some kind of message',
              substitutions: [],
            },
            stack: [],
            category: 'Some kind of message',
            componentStack: [],
          }),
        ]}
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render the latest warning notification', async () => {
    const output = await render.create(
      <LogBoxNotificationContainer
        selectedLogIndex={-1}
        logs={[
          new LogBoxLog({
            level: 'warn',
            isComponentError: false,
            message: {
              content: 'Some kind of message',
              substitutions: [],
            },
            stack: [],
            category: 'Some kind of message',
            componentStack: [],
          }),
          new LogBoxLog({
            level: 'warn',
            isComponentError: false,
            message: {
              content: 'Some kind of message (latest)',
              substitutions: [],
            },
            stack: [],
            category: 'Some kind of message (latest)',
            componentStack: [],
          }),
        ]}
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render the latest error notification', async () => {
    const output = await render.create(
      <LogBoxNotificationContainer
        selectedLogIndex={-1}
        logs={[
          new LogBoxLog({
            level: 'error',
            isComponentError: false,
            message: {
              content: 'Some kind of message',
              substitutions: [],
            },
            stack: [],
            category: 'Some kind of message',
            componentStack: [],
          }),
          new LogBoxLog({
            level: 'error',
            isComponentError: false,
            message: {
              content: 'Some kind of message (latest)',
              substitutions: [],
            },
            stack: [],
            category: 'Some kind of message (latest)',
            componentStack: [],
          }),
        ]}
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render both an error and warning notification', async () => {
    const output = await render.create(
      <LogBoxNotificationContainer
        selectedLogIndex={-1}
        logs={[
          new LogBoxLog({
            level: 'warn',
            isComponentError: false,
            message: {
              content: 'Some kind of message',
              substitutions: [],
            },
            stack: [],
            category: 'Some kind of message',
            componentStack: [],
          }),
          new LogBoxLog({
            level: 'error',
            isComponentError: false,
            message: {
              content: 'Some kind of message (latest)',
              substitutions: [],
            },
            stack: [],
            category: 'Some kind of message (latest)',
            componentStack: [],
          }),
        ]}
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render selected fatal error even when disabled', async () => {
    const output = await render.create(
      <LogBoxNotificationContainer
        isDisabled
        selectedLogIndex={0}
        logs={[
          new LogBoxLog({
            level: 'fatal',
            isComponentError: false,
            message: {
              content: 'Should be selected',
              substitutions: [],
            },
            stack: [],
            category: 'Some kind of message',
            componentStack: [],
          }),
        ]}
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render selected syntax error even when disabled', async () => {
    const output = await render.create(
      <LogBoxNotificationContainer
        isDisabled
        selectedLogIndex={0}
        logs={[
          new LogBoxLog({
            level: 'syntax',
            isComponentError: false,
            message: {
              content: 'Should be selected',
              substitutions: [],
            },
            stack: [],
            category: 'Some kind of syntax error message',
            componentStack: [],
            codeFrame: {
              fileName: '/path/to/RKJSModules/Apps/CrashReact/CrashReactApp.js',
              location: {row: 199, column: 0},
              content: `  197 | });
  198 |
> 199 | export default CrashReactApp;
      | ^
  200 |`,
            },
          }),
        ]}
      />,
    );

    expect(output).toMatchSnapshot();
  });
});
