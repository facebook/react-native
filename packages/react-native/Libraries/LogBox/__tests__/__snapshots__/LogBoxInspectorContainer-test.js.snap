// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogBoxNotificationContainer should render both an error and warning notification 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "bottom": 20,
      "left": 10,
      "position": "absolute",
      "right": 10,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 8,
        "marginBottom": 5,
        "overflow": "hidden",
      }
    }
  >
    <LogBoxLogNotification
      level="warn"
      log={
        LogBoxLog {
          "category": "Some kind of message",
          "codeFrame": undefined,
          "componentStack": Array [],
          "componentStackType": "legacy",
          "count": 1,
          "extraData": undefined,
          "isComponentError": false,
          "level": "warn",
          "message": Object {
            "content": "Some kind of message",
            "substitutions": Array [],
          },
          "onNotificationPress": undefined,
          "stack": Array [],
          "symbolicated": Object {
            "error": null,
            "stack": null,
            "status": "NONE",
          },
          "symbolicatedComponentStack": Object {
            "componentStack": null,
            "error": null,
            "status": "NONE",
          },
          "type": undefined,
        }
      }
      onPressDismiss={[Function]}
      onPressOpen={[Function]}
      totalLogCount={1}
    />
  </View>
  <View
    style={
      Object {
        "borderRadius": 8,
        "marginBottom": 5,
        "overflow": "hidden",
      }
    }
  >
    <LogBoxLogNotification
      level="error"
      log={
        LogBoxLog {
          "category": "Some kind of message (latest)",
          "codeFrame": undefined,
          "componentStack": Array [],
          "componentStackType": "legacy",
          "count": 1,
          "extraData": undefined,
          "isComponentError": false,
          "level": "error",
          "message": Object {
            "content": "Some kind of message (latest)",
            "substitutions": Array [],
          },
          "onNotificationPress": undefined,
          "stack": Array [],
          "symbolicated": Object {
            "error": null,
            "stack": null,
            "status": "NONE",
          },
          "symbolicatedComponentStack": Object {
            "componentStack": null,
            "error": null,
            "status": "NONE",
          },
          "type": undefined,
        }
      }
      onPressDismiss={[Function]}
      onPressOpen={[Function]}
      totalLogCount={1}
    />
  </View>
</RCTSafeAreaView>
`;

exports[`LogBoxNotificationContainer should render null with no logs 1`] = `null`;

exports[`LogBoxNotificationContainer should render null with no selected log and disabled 1`] = `null`;

exports[`LogBoxNotificationContainer should render selected fatal error even when disabled 1`] = `null`;

exports[`LogBoxNotificationContainer should render selected syntax error even when disabled 1`] = `null`;

exports[`LogBoxNotificationContainer should render the latest error notification 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "bottom": 20,
      "left": 10,
      "position": "absolute",
      "right": 10,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 8,
        "marginBottom": 5,
        "overflow": "hidden",
      }
    }
  >
    <LogBoxLogNotification
      level="error"
      log={
        LogBoxLog {
          "category": "Some kind of message (latest)",
          "codeFrame": undefined,
          "componentStack": Array [],
          "componentStackType": "legacy",
          "count": 1,
          "extraData": undefined,
          "isComponentError": false,
          "level": "error",
          "message": Object {
            "content": "Some kind of message (latest)",
            "substitutions": Array [],
          },
          "onNotificationPress": undefined,
          "stack": Array [],
          "symbolicated": Object {
            "error": null,
            "stack": null,
            "status": "NONE",
          },
          "symbolicatedComponentStack": Object {
            "componentStack": null,
            "error": null,
            "status": "NONE",
          },
          "type": undefined,
        }
      }
      onPressDismiss={[Function]}
      onPressOpen={[Function]}
      totalLogCount={2}
    />
  </View>
</RCTSafeAreaView>
`;

exports[`LogBoxNotificationContainer should render the latest warning notification 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "bottom": 20,
      "left": 10,
      "position": "absolute",
      "right": 10,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 8,
        "marginBottom": 5,
        "overflow": "hidden",
      }
    }
  >
    <LogBoxLogNotification
      level="warn"
      log={
        LogBoxLog {
          "category": "Some kind of message (latest)",
          "codeFrame": undefined,
          "componentStack": Array [],
          "componentStackType": "legacy",
          "count": 1,
          "extraData": undefined,
          "isComponentError": false,
          "level": "warn",
          "message": Object {
            "content": "Some kind of message (latest)",
            "substitutions": Array [],
          },
          "onNotificationPress": undefined,
          "stack": Array [],
          "symbolicated": Object {
            "error": null,
            "stack": null,
            "status": "NONE",
          },
          "symbolicatedComponentStack": Object {
            "componentStack": null,
            "error": null,
            "status": "NONE",
          },
          "type": undefined,
        }
      }
      onPressDismiss={[Function]}
      onPressOpen={[Function]}
      totalLogCount={2}
    />
  </View>
</RCTSafeAreaView>
`;
